pandas/tests/test_window.py

```diff
@@ -275,6 +275,17 @@ def test_how_compat(self):
                             getattr(s, t)(freq='D', **kwargs), op)(how=how)
                         assert_series_equal(result, expected)
 
+    # GH #12373 : rolling functions error on float32 data
+    def test_ensure_type_float64(self):
+        expected = pd.Series([np.nan, np.nan, 2, 3, 4])
+
```

Comment:
this should be very explict

```
for dtype in ['np.int64','np.int32','np.int16','np.int8',......'np.float64','np.float32'....]:
     ....
```

etc....this should test all int,float,uint types as well as ``object``.
assert that it is correctly raising on ``M8[ns],m8[ns],datetime64[ns, UTC],category``

there might be other tests for this already, if so, pls move these nearby.

Further should loop thru all operations (simply make a list of them)